src/App.tsx

```diff
@@ -5,7 +5,14 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const addMovie = (newMovie: Movie) => {
+    if (!movies.some(movie => movie.imdbId === newMovie.imdbId)) {
```

Comment:
move result of some method to a separate variable and use variable inside `if`
Keep your code as simple as possible, with variables - it's easier to understand what's going on in your code